lib/internal/error_serdes.js

```diff
@@ -52,7 +60,13 @@ function TryGetAllProperties(object, target = object) {
         // Continue regardless of error.
       }
     }
-    if ('value' in descriptor && typeof descriptor.value !== 'function') {
+    if (key === 'cause') {
+      delete descriptor.get;
```

Comment:
What if the `cause` property is defined as a getter?

Reply:
that is handled by https://github.com/MoLow/node/blob/b814c8c52e7e38eb351dcbe930e360c2d94e9e17/lib/internal/error_serdes.js#L55-L61
but I have added a test for it

Reply:
Can you add a test with a getter that throws please?

Reply:
Generally not a fan of using `delete` as it has very real performance implications.